### DISCUSSION TITLE
Using FDS to calculate the wind-induced force of flat roof

### DISCUSSION THREAD

**Discussion by Daniel931223** [firemodels/fds#12897] [Q&A]
Hello friends,

I am a beginner in FDS and want to calculate the roof lift caused by wind field. I set up "WALL PRESSURE" DEVICE on both outside and inside of building, i.e., the top and bottom of roof. If the area of roof is A, can the resultant wind-induced force be calculated as F=(P_outside + P_inside)*A? If I get the pressure -45Pa from the DEVICE at the top of roof, where should the direction of this pressure be directed?

BTW, when I simulate the wind field blowing an enclosed building, I find the pressure inside the building fluctuates as well. Why this happen? 

I attach .fds file for checking.
[TEST3.txt](https://github.com/firemodels/fds/files/15244025/TEST3.txt)

Any advice and comments are greatly appreciated!
D.S

**Comment by rmcdermo** [comment]
Theoretically, what you are attempting is possible, but I have not seen a validation study with FDS looking at pressure differentials across walls like this.  There has been some work looking at pressure coefficients on buildings, see `fds/Validation/UWO_Wind_Tunnel` and there has been work looking at interior pressure zones with a fire, see `fds/Validation/PRISME` and `fds/Validation/DelCo_Trainers`.

The lift force will be the pressure *difference* times the area.  If the pressure on top of the roof is -45 Pa and the pressure inside is 10 Pa, then the dP = 10 - (-45) = 55 Pa.

Your geometry is a sealed pressure zone.  But FDS starts out with small velocity fluctuations that mimic typical ambient air fluctuations (air is never completely still).  The pressure inside the zone should not change much, but with the default pressure solver you can get penetration errors with the immersed boundary method.  There is a discussion on how to handle this in user's guide, see the section on 21.1.1 Optional Pressure Solvers.  I suggest you use ULMAT here.
```
&PRES SOLVER='ULMAT'/
```
This will completely eliminate the solid boundary penetration errors, at a modest increase in cpu cost.  For very small pressure fluctuations, on the order of 10s of Pa, you cannot really tolerate any penetration errors.

**Reply by drjfloyd** [reply]
If you are running this with MPI you have 14 meshes that are 50,30,30 and 1 mesh with 200,240,120.  The 50,30,30 mesh has 45,000 grid cells, and the 200,240,120 mesh has 5.76 million grid cells. For MPI you want similar numbers of cells per mesh. WIth this input, cores assigned the 50,30,03 meshes will spend 99 % of their time waiting for the 200,240,120 mesh to complete a timestep.
